### ISSUE TITLE
Arabic recitation with English translation

### ISSUE THREAD

**Issue by spjps2009** [quran/quran_android#641]
Asalamualikum wa rahmatullah. 
Firstly Jazakallah khait for amazing Quran app
I was thinking it would be great to have feature that there can be Arabic recitation of ayah, then following that the English translation audio since many people don't understand it and would be useful for people listening in a car or places where they can't read translation. 

Jazakallah khair

**Comment by ozbek** [comment]
Wa alaikum assalam wa rahmatullah.

I also have been thinking about this for a while already. This probably can be achieved more easily using TTS for translations... 

Let's see what @ahmedre and others have to say about it.

**Comment by asimmohiuddin** [comment]
One way can also be to allow playback for two Qaris for each verse one after the other. Since there is Ibrahim Walk English translation, people can select him as the second reader. However, the problem with this would be that the English translation may not match the version the user has downloaded.

TTS sounds interesting. If it is natural enough it could work for different languages as well. However, I don't think it will get the appropriate inflections in speech so it might be difficult to understand. I don't know, what do you guys think? @ozbek @ahmedre

Lastly, there are timing files for Shuraim/sudais and pickthall and also Abdullah Basfar and Ibrahim walk readily available. These can be quickly implemented. Theoretically more reader combinations, audio, and timing files can be generated relatively easily if needed. The obvious problem with this approach is that people would need to download separate audio for each combination. For example, if we used the basfar and walk files available, people who might already have both basfar and walk separately need to download new audio for the combo.

**Comment by ozbek** [comment]
> However, I don't think it will get the appropriate inflections in speech so it might be difficult to understand.

Although natural inflections are not yet supported, TTS engines have pretty decent performance in some languages. English synthesis is understandable, for example.

**Comment by asimmohiuddin** [comment]
Yeah your are right it sounds pretty good. I think there was recently an update that improved it even more (Google TTS).

**Comment by spjps2009** [comment]
I think easiest way would be to do using two audios. One of a Reciter and the other of Ibrahim Walk and play one ayah of reciter then pause, then play translation then pause. etc.

**Comment by ahmedre** [comment]
jazakumAllah khairan - i just saw a video today about Google improving inflection and what not in google voice [here](http://www.theverge.com/2016/3/30/11333524/google-now-voice-improved-smoother-sound).

i think maybe we should consider doing both approaches - TTS is great if it works well since it opens the door for supporting languages in which we don't have mp3s of a translator. playing one of each is the straight forward approach, but right now, only Ibrahim Walk + [sheikh reciting Arabic] is what we'd have (there are a handful of translation verse by verse files on everyayah we can also use that are their respective language only as well).

**Comment by shakram02** [comment]
Assalamu alaikum, I was working on a [project](https://www.youtube.com/watch?v=4d8WVXH2p5c&list=PLO96dbwBSxCto1dcPhxs63D1kYezD1wN2) to make videos that display Arabic and translated text that's synced with audio during recitation. 

I can start working on adding this feature to the app isA. I already made a very basic prototype of the implementation [here](https://github.com/shakram02/quran_android/tree/testing-text-recitation), Highlight an Ayah and you'll find a new item with a tick icon in the AyahToolbar. When you click it just hit play. Sound won't be played as it's not implemented but it's for `Abdul_Basit_Murattal_192kbps` recitation from VerseByVerse website. It will match [this](https://youtu.be/xrCfANFur_w) video.

The dataset for the whole Quran is available for sh. Mishary Rashed and Abdulbasit, Quran-align is used for Abdulbasit and the next sheikhs to come isA (basically those provided by quran-align)

the [translation](https://quranenc.com/en/browse/spanish_garcia) used is that of Muhammad Isa Garcia from quranenc (you already provide it in Quran app)

What should be done next? jazakum Allah Kheir.
